mem0/memory/neptune_memory.py

```diff
@@ -0,0 +1,375 @@
+        """
+        Returns the OpenCypher query and parameters to search for source nodes
+
+        :param source_embedding: source vector
+        :param user_id: user_id to use
+        :param threshold: the threshold for similarity
+        :return: str, dict
+        """
+        cypher = """
+            MATCH (source_candidate)
+            WHERE source_candidate.embedding IS NOT NULL 
+            AND source_candidate.user_id = $user_id
+
+            WITH source_candidate, source_candidate.embedding AS embedding, $source_embedding as v_embedding
+            CALL neptune.algo.vectors.distanceByEmbedding(
+                embedding, 
+                source_candidate,
+                {metric:"CosineSimilarity"}
+            ) YIELD distance
+            WITH source_candidate, distance AS cosine_similarity
+            WHERE cosine_similarity >= $threshold
+
+            WITH source_candidate, cosine_similarity
+            ORDER BY cosine_similarity DESC
+            LIMIT 1
+
+            RETURN id(source_candidate), cosine_similarity
+            """
+
+        params = {
+            "source_embedding": source_embedding,
+            "user_id": user_id,
+            "threshold": threshold,
+        }
+        logger.debug(f"_search_source_node\n  query={cypher}")
+        return cypher, params
+
+    def _search_destination_node_cypher(self, destination_embedding, user_id, threshold):
+        """
+        Returns the OpenCypher query and parameters to search for destination nodes
+
+        :param source_embedding: source vector
+        :param user_id: user_id to use
+        :param threshold: the threshold for similarity
+        :return: str, dict
+        """
+        cypher = """
+                MATCH (destination_candidate)
+                WHERE destination_candidate.embedding IS NOT NULL 
+                AND destination_candidate.user_id = $user_id
+    
+                WITH destination_candidate, destination_candidate.embedding AS embedding, $destination_embedding as v_embedding
+                CALL neptune.algo.vectors.distanceByEmbedding(
+                    embedding, 
+                    destination_candidate,
+                    {metric:"CosineSimilarity"}
+                ) YIELD distance
+                WITH destination_candidate, distance AS cosine_similarity
+                WHERE cosine_similarity >= $threshold
+    
+                WITH destination_candidate, cosine_similarity
+                ORDER BY cosine_similarity DESC
+                LIMIT 1
+    
+                RETURN id(destination_candidate), cosine_similarity
+                """
+        params = {
+            "destination_embedding": destination_embedding,
+            "user_id": user_id,
+            "threshold": threshold,
+        }
+
+        logger.debug(f"_search_destination_node\n  query={cypher}")
+        return cypher, params
+
+    def _delete_all_cypher(self, filters):
```

Comment:
How is this going to handle issues when deleting large graphs?  Do we need to do this in batches?